### ISSUE TITLE
Create artifacts for feature branches (and maybe main)

### ISSUE THREAD

**Issue by Baklap4** [bohning/usdb_syncer#285]
Currently USDB_Syncer has a workflow set up which runs during tagging to create an actual release based on the version specified in the pyproject.toml

For windows this release ends up in a file called `USDB_Syncer-0.7.0-Windows.exe` (for the latest stable release).

### Feature requested
I'd love to introduce either alpha and/or beta 'versions'  depending on either a PR (alpha) or when merged into main (beta). This can be quite useful for testers of the application. This way they don't need to have the whole developer toolchain up and running but can rather just download the artifact in the readme as latest/nightly build or from the comment section of an open PR testing a specific new feature or bugfix.

When we started doing this at Performous we saw more traction and faster feedback cycles with the testers.

### Suggested implementation

* When creating a feature branch and pushing the branch create a new version: 0.8.0-alpha-commitsha
    * This results in a comment being posted on the PR where one could download the linux, mac and windows binary
* When the said feature is merged into main we can create a version: 0.8.0-beta-commitsha
    * This results in a downloadable artifacts which can be downloaded from the README.md
    * Every new merge results into a new binary being the latest.
* Once testing is done and all features are submitted for 0.8.0 create a tag and release the final binary through a releasenote.
    * This creates artifacts and a release page like it's already been done

I'd like to hear your feedback
